datasets/in/mha_banned/crawler.py

```diff
@@ -16,37 +16,53 @@ def parse_names(field: str) -> List[str]:
     return names
 
 
-def assert_link_hash(context: Context, doc: _Element, label: str, expected: str) -> str:
+def assert_link_hash(
+    context: Context,
+    doc: _Element,
+    label: str,
+    expected: str,
+    xpath: Optional[str] = None,
+) -> str:
     label_xpath = f".//td[contains(text(), '{label}')]"
     label_cells = doc.xpath(label_xpath)
     assert len(label_cells) == 1
     anchors = label_cells[0].xpath("./following-sibling::td//a")
     assert len(anchors) == 1
     link = anchors[0]
     url = link.get("href")
-    h.assert_url_hash(context, url, expected)
-    return url
+    if xpath:
+        success = h.assert_html_url_hash(context, url, expected, path=xpath)
+        if not success:
+            print(f"Hash assertion for {url} with XPath {xpath} failed")
+    else:
+        success = h.assert_url_hash(context, url, expected)
+        if not success:
+            print(f"Hash assertion for {url} failed")
+
+    return url if success else None
 
 
 def crawl(context: Context) -> None:
     doc = context.fetch_html(context.dataset.url)
     doc.make_links_absolute(context.dataset.url)
-    expected_sources: Set[str] = set()
+    expected_sources: Set[bool] = set()
 
     expected_sources.add(
         assert_link_hash(
             context,
             doc,
             "UNLAWFUL ASSOCIATIONS UNDER SECTION 3 OF UNLAWFUL ACTIVITIES (PREVENTION) ACT, 1967",
-            "582fd09cb946aa1bfc186da33d43642fa160cb3c",
+            "50cbbee5d7777188e5fd8d2e5b74ddcc6b537716",
+            xpath='.//div[@id="block-mhanew-content"]//div[3]//table',
```

Comment:
index-based xpath is a red flag. It's brittle, i.e. it could easily stop matching if they make uninteresting content changes, and I don't think it's contributing anything to the selection. It's not like you need to say "the table in the third div in some list of divs under block-mha-new-content, right? Is there more than one table in block-mhanew-content?

```suggestion
            xpath='.//div[@id="block-mhanew-content"]//table',
```